crates/entities/src/models/crawl_queue.rs

```diff
@@ -367,46 +369,64 @@ pub async fn enqueue_all(
         })
         .collect();
 
+    // Ignore urls already indexed
+    let mut is_indexed: HashSet<String> = HashSet::with_capacity(urls.len());
     // Igore urls already indexed
-    let is_indexed: HashSet<String> = indexed_document::Entity::find()
-        .filter(indexed_document::Column::Url.is_in(urls.clone()))
-        .all(db)
-        .await?
-        .iter()
-        .map(|x| x.url.to_string())
-        .collect();
+    for chunk in urls.chunks(10000) {
+        let chunk = chunk.iter().map(|url| url.to_string()).collect::<Vec<_>>();
+        for entry in indexed_document::Entity::find()
+            .filter(indexed_document::Column::Url.is_in(chunk.clone()))
+            .all(db)
+            .await?
+            .iter()
+        {
+            is_indexed.insert(entry.url.to_string());
+        }
+    }
 
     let to_add: Vec<ActiveModel> = urls
         .into_iter()
-        .filter(|url| !is_indexed.contains(url))
-        .map(|url| {
-            let parsed = Url::parse(&url).unwrap();
-            let domain = parsed.host_str().unwrap();
-
-            ActiveModel {
-                domain: Set(domain.to_string()),
-                crawl_type: Set(overrides.crawl_type.clone()),
-                url: Set(url),
-                ..Default::default()
+        .filter_map(|url| {
+            let mut result = None;
+            if !is_indexed.contains(&url) {
+                if let Ok(parsed) = Url::parse(&url) {
+                    if let Some(domain) = parsed.host_str() {
+                        result = Some(ActiveModel {
+                            domain: Set(domain.to_string()),
+                            crawl_type: Set(overrides.crawl_type.clone()),
+                            url: Set(url.to_string()),
+                            ..Default::default()
+                        });
+                    }
+                }
             }
+            result
         })
         .collect();
 
     if to_add.is_empty() {
         return Ok(());
     }
-
-    let (sql, values) = Entity::insert_many(to_add)
-        .query()
-        .on_conflict(OnConflict::column(Column::Url).do_nothing().to_owned())
-        .build(SqliteQueryBuilder);
-
-    let values: Vec<Value> = values.iter()
-        .map(|x| x.to_owned())
-        .collect();
-    match db.execute(Statement::from_sql_and_values(DbBackend::Sqlite, &sql, values)).await {
-        Ok(_) => {}
-        Err(e) => log::error!("insert_many error: {:?}", e),
+    for to_add in to_add.chunks(1000) {
```

Comment:
Any reason why this is chunked to 1000 and the one above is to 10,000?

Reply:
No real evidence. I was thinking the number would be smaller after filtered. There is "more" data sent. I had smaller batches because of the dup in the list before. that was before i did the sort dedup and merged in the on conflict change.